packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/cluster.ts

```diff
@@ -129,6 +129,52 @@ export class ClusterResourceHandler extends ResourceHandler {
       return this.onCreate();
     }
 
+    // Update tags
+    if (updates.updateTags) {
+      try {
+        //Describe cluster to get arn and to make sure Cluster exists
+        const cluster = (await this.eks.describeCluster({ name: this.clusterName })).cluster;
+        if (this.oldProps.tags) {
+          if (this.newProps.tags) {
+            //Update tags
+            const tagsToAdd = getTagsToUpdate(this.oldProps.tags, this.newProps.tags);
+            if (tagsToAdd) {
+              const tagConfig: EKS.TagResourceCommandInput = {
+                resourceArn: cluster?.arn,
+                tags: tagsToAdd,
+              };
+              await this.eks.tagResource(tagConfig);
+            }
+            //Remove tags
+            const tagsToRemove = getTagsToRemove(this.oldProps.tags, this.newProps.tags);
+            if (tagsToRemove.length > 0 ) {
+              const config: EKS.UntagResourceCommandInput = {
+                resourceArn: cluster?.arn,
+                tagKeys: tagsToRemove,
+              };
+              await this.eks.untagResource(config);
+            }
+          } else {
+            //Remove all tags
```

Comment:
why are we removing all the tags here?

Reply:
This is the else where `newtags` would be empty hence we have to remove all the `oldtags` from the cluster

Reply:
to make sure I understood correctly, newtags can be empty in case where customer doesn't want to introduce any change to the existing ones eg. looking at the condition here, in case tags under newProps come as empty would this remove all the existing ones ?
`updateTags: JSON.stringify(newProps.tags) !== JSON.stringify(oldProps.tags)`

Reply:
>newtags can be empty in case where customer doesn't want to introduce any change to the existing ones

In that case, `newTags` and `oldTags` would be same. `newTags` would be empty if customer wish to remove all tags from the cluster. 

`newTags` represent the desired state that we want the cluster tags to be in post the update. It will be the same as compared to `oldTags` if there has been no changes to tags done in CDK code.

Reply:
Got it, thank you for explaning